src/sentry/api/serializers/models/discoversavedquery.py

```diff
@@ -38,6 +40,13 @@ def serialize(self, obj, attrs, user, **kwargs):
             if obj.query.get(key) is not None:
                 data[key] = obj.query[key]
 
+        # expire queries that are beyond the retention period
+        if "start" in obj.query:
+            start, end = parse_timestamp(obj.query["start"]), parse_timestamp(obj.query["end"])
```

Comment:
There are datetime strings generated from legacy Discovery queries (https://github.com/getsentry/sentry/pull/16699) that can error at this line:

```
>>> parse_timestamp("2019-10-16T19:31:50+0000") is None
True
```

Specifically, `start` and/or `end` may be `None`.

Reply:
@wmak you mentioned that I didn't need to worry about older versions of Discovery queries, is this what you were referring to?